src/schema/posts.ts

```diff
@@ -1101,6 +1223,117 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
   BaseContext
 >({
   Query: {
+    sourcePostModeration: async (
+      _,
+      { id, sourceId }: { id: string; sourceId: string },
+      ctx: Context,
+      info,
+    ): Promise<GQLSourcePostModeration> => {
+      await ensureSourcePermissions(ctx, sourceId);
+
+      return graphorm.queryOneOrFail<GQLSourcePostModeration>(
+        ctx,
+        info,
+        (builder) => ({
+          ...builder,
+          queryBuilder: builder.queryBuilder.where(
+            `"${builder.alias}"."id" = :id AND ${builder.alias}."sourceId" = :sourceId`,
+            { id, sourceId },
+          ),
+        }),
+      );
+    },
+    sourcePostModerations: async (
+      _,
+      args: SourcePostModerationArgs,
+      ctx: Context,
+      info,
+    ): Promise<ConnectionRelay<GQLSourcePostModeration>> => {
+      const isModerator = await isPrivilegedMember(ctx, args.sourceId);
```

Comment:
I think we need sourcePermissions here as well right?

Reply:
No. The isPrivilegedMember function gets the source user by using userId + sourceId, so it will only get members of the source. I should modify the findOneBy to findOneByOrFail, so that if you're not a member at all it should error.

If it actually finds the user it checks if the user is of rank moderator or higher.

Reply:
If this will now also throw, this should be ensureSourcePermissions, if not let's leave it as is, it was brought by multiple times already in other PR

Reply:
Clarified in DMs 🤝

Reply:
🤝